website/docs/sql/datatype/array.md

```diff
@@ -0,0 +1,84 @@
+# Array Type
+
+An array is a sequential collection of elements packed into a single SQL value.
```

Comment:
We should align our documentation with the one from PostgreSQL where feasiable and where we agree with the PostgreSQL documentation. It doesn't need to be exact, but I think the PostgreSQL array page has quite some good contents we should copy.

Reply:
I'm not sure I agree. I found the PostgreSQL documentation to be quite obtuse and difficult to navigate. Particularly, my gripes with it are:

 - The audience for this documentation are probably not software engineers, but business-analyst-type roles. The PostgreSQL docs never explain what an array is; they also refer to `C` without specifying it as a programming language. I don't think this is the level-of-detail we're aiming for.
 - In many places, the PostgreSQL docs introduce some concept or syntax (e.g., size specifiers, dimension specifiers) only to mention in the next sentence that the current implementation ignores it. Admittedly, this is probably more a testament to PostgreSQL's array design, but is also, again, a very developer-like view. In my experience, users do not have internalized the notion of interface vs. implementation; and they should not have to care. I do not think PostgreSQL's documentation shines in this regard.
 - Having explanations and examples interleaved (as PostgreSQL does) makes a good first-time reading experience, but makes it hard to locate information. When implementing arrays, I personally had a hard time looking up certain things, as speculative `CTRL`-`F `yields many false positives with this prose-heavy style. I would estimate that most readers are looking for a reference, rather than a tutorial.

Could you please be more specific in what you like about the PostgreSQL documentation, and why?

Reply:
See [below](https://github.com/tableau/hyper-db/pull/110#discussion_r1704373872): 
[bda7bea](https://github.com/tableau/hyper-db/pull/110/commits/bda7bea6e20cbb11681f1ca50140f8258c77a809)